src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/OnTypeFormattingHandler.cs

```diff
@@ -71,31 +72,37 @@ public async Task<TextEdit[]> HandleRequestAsync(DocumentOnTypeFormattingParams
                 return null;
             }
 
-            // For onTypeFormatting, it makes more sense to look up the projection of the character that was inserted. Aka. request.Position.Character - 1
-            var position = new Position(request.Position.Line, request.Position.Character - 1);
```

Comment:
Doing this was wrong for cases like when going to the previous character would move to the previous line. Also, trigger character is a string and its length can be more than 1.